shapely/errors.py

```diff
@@ -1,5 +1,35 @@
 """Shapely errors."""
-from shapely.lib import GEOSException, ShapelyError  # NOQA
+import threading
+
+from shapely.lib import _setup_signal_checks, GEOSException, ShapelyError  # NOQA
+
+
+def setup_signal_checks(interval=10000, main_thread_id=None):
+    """This enables Python signal checks in the ufunc inner loops.
+
+    Doing so allows termination (using CTRL+C) of operations on large arrays of vectors.
+
+    Parameters
+    ----------
+    interval : int, default 10000
+        Check for interrupts every x iterations. The higher the number, the slower
+        shapely will respond to a signal. However, at low values there will be a negative effect
+        on performance. The default of 10000 does not have any measureable effects on performance.
+    main_thread_id : int, defaults to ``threading.get_ident``
+        Python signal handlers are always executed in the main Python thread of the main
+        interpreter. Shapely needs to know the main thread id to prevent needless signal
+        checks which would deteriorate performance in multithreading contexts.
```

Comment:
Are there use cases for specifying this manually?

Reply:
@caspervdw one question above

Reply:
I guess things go wrong if someone would `import shapely` for the first time in a subthread AND then try to break a shapely operation in a main thread. I just exposed it here because the C-level function exposes it (and that is because it is easier to get the current thread id from python).

So what do you say, should I remove this parameter?

Reply:
> I guess things go wrong if someone would `import shapely` for the first time in a subthread AND then try to break a shapely operation in a main thread.

Do you think it is reasonable in that case to set the id of the main thread manually to avoid that situation? (an easier alternative might be to ensure you import shapely in the main thread?)

> So what do you say, should I remove this parameter?

I was just wondering about a use case. If there are cases were it could be useful, it's certainly fine to keep it.

Reply:
I replaced the thread-id fetch with `threading.main_thread().ident` and removed the parameter. I think it is good to merge now.